### ISSUE TITLE
[otbn,spec] Is the SW team happy with our decision not to expose a stop address?

### ISSUE THREAD

**Issue by rswarbrick** [lowRISC/opentitan#7158]
The current OTBN design doesn't expose the PC of the final instruction that was executed. This is different from Ibex, where we're planning some sort of crash dump mechanism.

We *think* this is probably better from a security point of view, and it's definitely less work for HW and DV. But we thought we should make absolutely sure that the SW team knows and is ok with it.

In general, we expect the OTBN simulator to be a cycle-accurate model of the core, so we'd expect any bring-up problems to be reproducible there. We also already expose an instruction counter, so a developer could probably figure out exactly what instruction was at fault by running the code through the simulator and then looking for the correct line number(!).

Could someone from the SW side check that this plan looks ok? @moidx @cfrantz

> estimate 2

**Comment by GregAC** [comment]
Whilst model and HW behaviour should match precisely you could get a divergence if the initial dmem/imem aren't the same. E.g. say you initialise dmem with some initial arguments then your OTBN program reads from an uninitialized location accidentally then branches based upon it. If you fully initialise both 4k memories before every OTBN run this wouldn't be an issue (you could at least do it as a debugging option even if this wasn't the default).

There's also been some talk of an OTBN only part of dmem, which Ibex couldn't access so you couldn't replicate it's state to reproduce on the simulator. Practically I don't think this will be an issue as OTBN software should be writing to it before reading anyway and you could add a little checker to the simulator to warn if the program reads a byte from that area if it hasn't yet written that byte.

**Comment by imphil** [comment]
We talked this through today with @cfrantz, @cdgori, @moidx and others and the initial feeling is that even though having the stop address could be useful for debugging, but for many scenarios the same  debugging insight can be obtained by modifying the software that runs on OTBN to explicitly write out the necessary information.

We need to ensure, however, that the DMEM secure wipe isn't erasing this access path to debug information. I added a note at https://github.com/lowRISC/opentitan/issues/6955#issuecomment-878534560.

So for now let's close this issue as "we won't implement a stop address". We can revisit at any time if we have a clear need for such functionality.
